src/Compilers/CSharp/Test/Semantic/Semantics/SynthesizedStaticConstructorTests.cs

```diff
@@ -265,6 +265,24 @@ class C
             Assert.True(IsBeforeFieldInit(typeSymbol));
         }
 
+        [WorkItem(543606, "http://vstfdevdiv:8080/DevDiv2/DevDiv/_workitems/edit/543606")]
+        [Fact]
+        public void StaticConstructorNullInitializer()
+        {
+            var source = @"
+#nullable enable
+class C
+{
+    static string s1 = null!;
+}";
+
+            var typeSymbol = CompileAndExtractTypeSymbol(source);
+
+            // Although we do not emit the synthesized static constructor, the source type symbol will still appear to have one
+            Assert.True(HasSynthesizedStaticConstructor(typeSymbol));
+            Assert.True(IsBeforeFieldInit(typeSymbol));
```

Comment:
>IsBeforeFieldInit [](start = 24, length = 17)

Shouldn't this be false since we won't be emitting this constructor. We should have metadata equivalent to a situation when we didn't even synthesize the constructor symbol in the first place.

Reply:
I decided to keep existing behavior around when a synthesized static constructor symbol is included, because removing it  requires some churn around flow analysis of static initializers, etc. and it could be a breaking behavior change for outside consumers.

In addition, it seemed impractical to decide to include the synthesized static constructor symbol based on whether there are "meaningful" static initializers. That requires binding the initializers, which we want to avoid doing just for a call to TypeSymbol.GetMembers().

Reply:
I added a test which demonstrates that identical type IL is generated with or without the `null` initializer.

Reply:
> I added a test which demonstrates that identical type IL is generated with or without the null initializer.

Where is the test and what is its name?

---
In reply to: [411767188](https://github.com/dotnet/roslyn/pull/43234#discussion_r411767188) [](ancestors = 411767188)

Reply:
The test `StaticNullInitializerHasNoEffectOnTypeIL` in CodeGenConstructorInitTests.cs